rasterio/transform.py

```diff
@@ -439,19 +439,19 @@ def xy(self, rows, cols, zs=None, offset='center'):
         temp_rows = []
         temp_cols = []
         try:
-            for pt in zip(cols, rows):
-                y, x = T * pt
-                temp_rows.append(y)
-                temp_cols.append(x)
+            for colrow in zip(cols, rows):
+                col, row = T * colrow
+                temp_rows.append(row)
+                temp_cols.append(col)
 
-            new_ys, new_xs = self._transform(
-                temp_rows, temp_cols, zs, transform_direction=TransformDirection.forward
+            new_xs, new_ys = self._transform(
+                temp_cols, temp_rows, zs, transform_direction=TransformDirection.forward
             )
 
-            if len(new_ys) == 1 and not AS_ARR:
-                return (new_ys[0], new_xs[0])
+            if len(new_xs) == 1 and not AS_ARR:
+                return (new_xs[0], new_ys[0])
             else:
-                return (new_ys, new_xs)
+                return (new_xs, new_ys)
```

Comment:
I'm a bit confuse. Those changes doesn't make things clearer for me. I think because we have rows/cols as input we should have output variable names to be `new_rows` / `new_cols`. 

Or we go all in and rename the input x/y which will make more senses considering the function's name

Reply:
From what I can tell, the input is rows and columns and the output is x,y. The change made here is the order of x and y on the output. I think that the order was incorrect when naming the internal variables originally. Do you agree with the change in order?

Reply:
Yes, that is what caused me do make this change. I changed 443 to col, row instead of x, y because it looks to me like it is just the col, row with the offset applied and hasn't been converted to x,y coordinates yet. Does that seem correct?

Reply:
yeah, it seems correct, I'm just not really sure with the variable naming thought the function `xy` (from the function's name) -> `row/col` (in the doctoring) -> `row/cols` (line 443) -> `xy` (output defined at line 447) .

Reply:
Or we could use the `offset` prefix instead of `shifted`.

Reply:
`new_rows, new_cols` is fine too I believe

Reply:
> new_rows, new_cols is fine too I believe

After pushing the commit, I am realizing that the original name was `temp_rows|cols` and am understanding that you are suggesting them to be named `new_rows|cols`. Not sure how I missed that earlier. I think that would work as well. I have pushed up `offset_rows|cols` and hopefully that helps the code be clear as well.